packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -214,18 +217,29 @@ function getValidTelemetryProps(obj: any): ITelemetryProperties {
 }
 
 /**
- * Helper class for error tracking that can be used to log an error in telemetry.
- * The props passed in (and any set directly on the object after the fact) will be
- * logged in accordance with the given tag, if present.
+ * Base class for "trusted" errors we create, whose properties can generally be logged to telemetry safely.
+ * All properties set on the object, or passed in (via the constructor or getTelemetryProperties),
+ * will be logged in accordance with their tag, if present.
  *
- * PLEASE take care to properly tag properties set on this object
+ * PLEASE take care to avoid setting sensitive data on this object without proper tagging!
  */
 export class LoggingError extends Error implements ILoggingError {
+    /**
+     * Create a new LoggingError
+     * @param message - Error message to use for Error base class
+     * @param props - telemetry props to include on the error for when it's logged
+     * @param omitPropsFromLogging - properties by name to omit from telemetry props
+     */
     constructor(
         message: string,
         props?: ITelemetryProperties,
+        private readonly omitPropsFromLogging: Set<string> = new Set(),
```

Comment:
No sure this will work, but you could try
```suggestion
        private readonly omitPropsFromLogging: Set<keyof this> = new Set(),
```

Reply:
Yeah I had hoped that would work too but it doesn't